File: shared/main/java/net/vitox/particle/util/RenderUtils.java
package net.vitox.particle.util;

import static org.lwjgl.opengl.GL11.*;

import net.ccbluex.liquidbounce.utils.MinecraftInstance;

public final class RenderUtils
{

	public static void connectPoints(final float xOne, final float yOne, final float xTwo, final float yTwo)
	{
		glPushMatrix();
		glEnable(GL_LINE_SMOOTH);
		glColor4f(1.0F, 1.0F, 1.0F, 0.8F);
		glDisable(GL_TEXTURE_2D);
		glBlendFunc(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
		glEnable(GL_BLEND);
		glLineWidth(0.5F);
		glBegin(GL_LINES);
		glVertex2f(xOne, yOne);
		glVertex2f(xTwo, yTwo);
		glEnd();
		glColor4f(1.0F, 1.0F, 1.0F, 1.0F);
		glDisable(GL_LINE_SMOOTH);
		glEnable(GL_TEXTURE_2D);
		glPopMatrix();
	}

	public static void drawCircle(final float x, final float y, final float radius, final int color)
	{
		final float alpha = (color >> 24 & 0xFF) / 255.0F;
		final float red = (color >> 16 & 0xFF) / 255.0F;
		final float green = (color >> 8 & 0xFF) / 255.0F;
		final float blue = (color & 0xFF) / 255.0F;

		final float floatPI = (float)Math.PI;

		glColor4f(red, green, blue, alpha);
		glEnable(GL_BLEND);
		glDisable(GL_TEXTURE_2D);
		glBlendFunc(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
		glEnable(GL_LINE_SMOOTH);
		glPushMatrix();
		glLineWidth(1.0F);
		glBegin(GL_POLYGON);
		for (int i = 0; i <= 360; i++)
			glVertex2d(x + MinecraftInstance.functions.sin(i * floatPI / 180.0F) * radius, y + MinecraftInstance.functions.cos(i * floatPI / 180.0F) * radius);
		glEnd();
		glPopMatrix();
		glEnable(GL_TEXTURE_2D);
		glDisable(GL_LINE_SMOOTH);
		glColor4f(1.0F, 1.0F, 1.0F, 1.0F);
	}

	private RenderUtils() {
	}
}

File: src/main/java/net/vitox/particle/util/RenderUtils.kt
package net.vitox.particle.util

import net.ccbluex.liquidbounce.utils.MinecraftInstance
import net.ccbluex.liquidbounce.utils.extensions.cos
import net.ccbluex.liquidbounce.utils.extensions.sin
import net.ccbluex.liquidbounce.utils.extensions.toRadians
import net.ccbluex.liquidbounce.utils.render.RenderUtils.glColor
import net.ccbluex.liquidbounce.utils.render.RenderUtils.resetColor
import org.lwjgl.opengl.GL11

object RenderUtils : MinecraftInstance()
{
    fun connectPoints(xOne: Float, yOne: Float, xTwo: Float, yTwo: Float)
    {
        GL11.glPushMatrix()
        GL11.glEnable(GL11.GL_LINE_SMOOTH)
        GL11.glColor4f(1.0f, 1.0f, 1.0f, 0.8f)
        GL11.glDisable(GL11.GL_TEXTURE_2D)
        GL11.glBlendFunc(GL11.GL_SRC_ALPHA, GL11.GL_ONE_MINUS_SRC_ALPHA)
        GL11.glEnable(GL11.GL_BLEND)
        GL11.glLineWidth(0.5f)

        GL11.glBegin(GL11.GL_LINES)
        GL11.glVertex2f(xOne, yOne)
        GL11.glVertex2f(xTwo, yTwo)
        GL11.glEnd()

        resetColor()
        GL11.glDisable(GL11.GL_LINE_SMOOTH)
        GL11.glEnable(GL11.GL_TEXTURE_2D)
        GL11.glPopMatrix()
    }

    fun drawCircle(x: Float, y: Float, radius: Float, color: Int)
    {
        glColor(color)
        GL11.glEnable(GL11.GL_BLEND)
        GL11.glDisable(GL11.GL_TEXTURE_2D)
        GL11.glBlendFunc(GL11.GL_SRC_ALPHA, GL11.GL_ONE_MINUS_SRC_ALPHA)
        GL11.glEnable(GL11.GL_LINE_SMOOTH)
        GL11.glPushMatrix()
        GL11.glLineWidth(1.0f)
        GL11.glBegin(GL11.GL_POLYGON)

        (0..360).map {
            val radians = it.toFloat().toRadians
            radians.sin to radians.cos
        }.forEach { GL11.glVertex2f(x + it.first * radius, y + it.second * radius) }

        GL11.glEnd()
        GL11.glPopMatrix()
        GL11.glEnable(GL11.GL_TEXTURE_2D)
        GL11.glDisable(GL11.GL_LINE_SMOOTH)
        resetColor()
    }
}
